### ISSUE TITLE
How to configure JGIT CredentialsProvider?

### ISSUE THREAD

**Issue by DieterVDW** [danielflower/multi-module-maven-release-plugin#21]
Hey this plugin looks awesome, but I can't get it working because I always get the following error:
`Authentication is required but no CredentialsProvider has been registered`

I guess I need to configure the credentials for JGIT to work. But I can't seem to find how to do this.
I find some info on programmatically configuring JGIT, but never in a Maven context.

I'm suspecting this is a dumb question, but however can't seem to find the answer...
Any tips?

**Comment by danielflower** [comment]
First off... are you using ssh or http git URLs? Presumably whatever you use requires a password. This isn't currently supported by the plugin sorry.

**Comment by cosimalaube** [comment]
I also struggle with configuring the CredentialsProvider. 
I have a freestyle Jenkins job with Git SCM configured with working SSH credentials. I call then the maven plugin and it seems to already use my SSH URL (from the Git SCM section) but somehow it cannot do the authentication.

Could you point me in the right direction?

I'd also appreciate a Jenkinsfile snippet if that's easier for you (as this is what I originally wanted to do... just switched to plain old GUI configuration to first get it working at all). Thanks a lot in advance!

**Comment by danielflower** [comment]
Hi, I don't have any jenkins config sorry, but I have a couple of questions....

1. What user does jenkins run as, and does that user have an id_rsa in their ~/.ssh/ directory?
2. Does the SSH key have a passphrase?

One way to investigate is to log in as the user and try to SSH to your git server, and see the keys used etc.

**Comment by cosimalaube** [comment]
Hi Daniel,  thank you for your quick response. 

regarding basic SSH stuff: I already double-checked SSH stuff (id_rsa in place, no passphrase) - that is fine. :-)

Now it seems that the used "jsch-agent-proxy" needs the "netcat" command (which I first did not have on the OS running in my Docker container). Added that and figured out that there seem to be different netcat versions (mine does not support "netcat -U" which seems to be needed).

**Comment by evaldeslacasa** [comment]
Hello Daniel. Nice plugin, thanks for your work.

I am experiencing the same issue when trying to run a release on Windows 10 in my local machine. My private key is in C:\Users\{$userName}\.ssh\id_rsa so I'm not sure why wouldn't it authenticate to GitHub. 

Any suggestion will be welcome.

Thanks again!

**Comment by cosimalaube** [comment]
Hello Enrique,

Have you tried figuring out if your netcat equivalent on Win10 works? 

I don't have neither running env nor sourcecode at hand but I remember that resolving my difficulties with authentication 10months ago boiled down to digging into the sourcecode, find out what Credentials Provider does/needs on my OS and then providing that (which was netcat in the proper version to support netcat -U on the Ubuntu OS I had back then).

Cosima

**Comment by danielflower** [comment]
Going to close this one due to its age, but feel free to re-open if there are still issues.

To answer the question in the title of this issue though, please see [SSH Authentication](https://danielflower.github.io/multi-module-maven-release-plugin/ssh-authentication.html) or [HTTPS Authentication](https://danielflower.github.io/multi-module-maven-release-plugin/https-authentication.html)

**Comment by evaldeslacasa** [comment]
Hello Daniel! I hope you are doing well!

It's been a while, but I had to face the same problem this week...

I followed the configuration guide for SSH and HTTPS:
[SSH Authentication](http://danielflower.github.io/multi-module-maven-release-plugin/ssh-authentication.html
)[HTTPS Authentication](http://danielflower.github.io/multi-module-maven-release-plugin/https-authentication.html)

Unfortunately I'm getting the following error:

```
[ERROR] There was an error while accessing the Git repository. The error returned from git was:
[ERROR] https://github.com/repollc/my-plugins: Authentication is required but no CredentialsProvider has been registered
[ERROR] Stack trace:
[ERROR] org.eclipse.jgit.api.errors.TransportException: https://github.com/repollc/my-plugins: Authentication is required but no CredentialsProvider has been registered
        at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:221)
        at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:159)
        at com.github.danielflower.mavenplugins.release.LocalGitRepo.allRemoteTags(LocalGitRepo.java:186)
        at com.github.danielflower.mavenplugins.release.Reactor.getRemoteBuildNumbers(Reactor.java:120)
        at com.github.danielflower.mavenplugins.release.Reactor.fromProjects(Reactor.java:46)
        at com.github.danielflower.mavenplugins.release.ReleaseMojo.execute(ReleaseMojo.java:114)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.eclipse.jgit.errors.TransportException: https://github.com/thirdwavellc/aoc-ui: Authentication is required but no CredentialsProvider has been registered
        at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:489)
        at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:290)
        at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:198)
        ... 27 more
```

So it looks like my authentication is having issues. By the way I tried this on Open PowerShell, Git Bash and Windows Subsystem for Linux (WSL 2). On WSL 2, I seem to have the `netcat -U` command working properly...

Any feedback will be appreciated!

Many thanks.

**Comment by evaldeslacasa** [comment]
Hello again Daniel!

We were able to resolve the issue...so there's no need to reopen the case.

The problem was that my Windows 10 didn't have my GitHub SSH key configured correctly. I had to make sure it was added in Pageant in a .ppk format. Once it was, the problem was gone. In case anyone ends up here with a similar issue, I recommend following: https://docs.acquia.com/cloud-platform/manage/ssh/sftp-key/.

Thank you for the plugin, it does a great job!
